packages/extension-ui/src/Popup/AuthManagement/AccountManagement.tsx

```diff
@@ -3,81 +3,140 @@
 
 import type { ThemeProps } from '../../types';
 
-import React, { useCallback, useContext, useEffect } from 'react';
-import { useParams } from 'react-router';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { RouteComponentProps, withRouter } from 'react-router';
 import styled from 'styled-components';
 
-import { AccountContext, ActionContext, Button } from '../../components';
+import { AuthUrls } from '@polkadot/extension-base/background/handlers/State';
+
+import {
+  AccountContext,
+  ActionContext,
+  Button,
+  ButtonArea,
+  FaviconBox,
+  RemoveAuth,
+  VerticalSpace
+} from '../../components';
+import Checkbox from '../../components/Checkbox';
 import useTranslation from '../../hooks/useTranslation';
-import { getAuthList, updateAuthorization } from '../../messaging';
+import { getAuthList, removeAuthorization, updateAuthorization } from '../../messaging';
 import { AccountSelection, Header } from '../../partials';
 
-interface Props extends ThemeProps {
+interface Props extends RouteComponentProps, ThemeProps {
   className?: string;
 }
 
-function AccountManagement({ className }: Props): React.ReactElement<Props> {
-  const { url } = useParams<{ url: string }>();
+const CustomButtonArea = styled(ButtonArea)`
+  padding-top: 16px;
+  padding-bottom: 0px;
+`;
+
+function AccountManagement({ className, location: { search } }: Props): React.ReactElement<Props> {
   const { selectedAccounts = [], setSelectedAccounts } = useContext(AccountContext);
   const { t } = useTranslation();
   const onAction = useContext(ActionContext);
+  const searchParams = new URLSearchParams(search);
+  const url = searchParams.get('url');
+
+  const [authList, setAuthList] = useState<AuthUrls | null>(null);
+
+  useEffect(() => {
+    getAuthList()
+      .then(({ list }) => setAuthList(list))
+      .catch((e) => console.error(e));
+  }, []);
+
+  const removeAuth = useCallback((url: string) => {
+    removeAuthorization(url)
+      .then(({ list }) => setAuthList(list))
+      .catch(console.error);
+  }, []);
+
+  const _removeAuth = useCallback(() => url && removeAuth(url), [removeAuth, url]);
 
   useEffect(() => {
     getAuthList()
       .then(({ list }) => {
-        if (!list[url]) {
+        if (url && !list[url]) {
           return;
         }
 
-        setSelectedAccounts && setSelectedAccounts(list[url].authorizedAccounts);
+        url && setSelectedAccounts && setSelectedAccounts(list[url].authorizedAccounts);
```

Comment:
I would prefer normal if when it's not assignment to a const just running a function.